### DISCUSSION TITLE
Is it possible to access frame buffer from cs-script?

### DISCUSSION THREAD

**Discussion by A-tG** [mpvnet-player/mpv.net#403] [Q&A] [answered]
I made csharp script that copy screenshot to clipboard without 3rd party dependecies https://github.com/A-tG/mpv.net-scripts/blob/main/scripts-cs/screenshot-to-clipboard.cs
But I dont like that it creates temporary file. Is there a way to get "screenshot" directly into a memory, without creating file?

**Comment by stax76** [comment]
There is a suitable command described here:

https://mpv.io/manual/master/#command-interface-screenshot-raw

But marshalling the data might be a challenge, once I tried such data marshalling and failed, maybe it's not even possible. It's easier to write in C and load a C library.

Or maybe a creative usage of VapourSynth would be possible.

https://mpv.io/manual/master/#video-filters-vapoursynth

**Comment by A-tG** [comment]
I know some stuff about marshaling, however I have no idea how to get result from "screenshot-raw" command (or how to call internal version of it), because obviously CommandV() return void

**Reply by stax76** [reply]
It's described here:

https://github.com/mpv-player/mpv/blob/master/libmpv/client.h#L933

**Reply by A-tG** [reply]
I think I moving into the right direction. I get 0 with this code https://github.com/A-tG/mpv.net-scripts/blob/85735fcfb4a4df0fa7e64a9f43008700d55ec661/scripts-cs/screenshot-to-clipboard.cs#L60
if I change command to something else (like "screenshot-raw1") I get -4, so its detecting command properly. ~~But result mpv_node seems not initialized at all, it have MPV_FORMAT_NONE format.~~

P.S. Fixed by using ref keyword

**Comment by A-tG** [comment, marked as answer]
Made it ~~https://github.com/A-tG/mpv.net-scripts/blob/main/scripts-cs/screenshot-to-clipboard.cs~~ https://github.com/A-tG/mpv.net-ScriptsExtension/blob/main/AtgScriptsExtension/ScreenshotToClipboardScript.cs
Quite ugly code. SetPixel() in a loop is very slow, and the whole thing probably can be optimized with usnafe code and allocating structures with stackalloc instead of AllocHGlobal()

**Reply by stax76** [reply]
Even though it's mpv.net specific, I would still tag the repo with `mpv` and also `mpv-script`.
